samcli/lib/delete/cfn_utils.py

```diff
@@ -121,11 +184,50 @@ def wait_for_delete(self, stack_name):
         try:
             waiter.wait(StackName=stack_name, WaiterConfig=waiter_config)
         except WaiterError as ex:
-            stack_status = ex.last_response.get("Stacks", [{}])[0].get("StackStatusReason", "")
+            stack_status = ex.last_response.get("Stacks", [{}])[0].get("StackStatusReason", "")  # type: ignore
 
             if "DELETE_FAILED" in str(ex):
                 raise CfDeleteFailedStatusError(
                     stack_name=stack_name, stack_status=stack_status, msg="ex: {0}".format(ex)
                 ) from ex
 
             raise DeleteFailedError(stack_name=stack_name, stack_status=stack_status, msg="ex: {0}".format(ex)) from ex
+
+    def _get_change_set_name(self, stack_name: str) -> str:
+        """
+        Returns the name of the change set for a stack
+
+        Parameters
+        ----------
+        stack_name: str
+            The name of the stack to find a change set
+
+        Returns
+        -------
+        str
+            The name of a change set
+
+        Raises
+        ------
+        FetchChangeSetError
+            Raised if there are boto call errors or parsing errors
+        NoChangeSetFoundError
+            Raised if a stack does not have any change sets
+        """
+        try:
+            change_sets: dict = self._client.list_change_sets(StackName=stack_name)
+        except (ClientError, BotoCoreError) as ex:
+            LOG.debug("Failed to perform boto call to fetch change sets")
+            raise FetchChangeSetError(stack_name=stack_name, msg=str(ex)) from ex
+
+        change_sets = change_sets.get("Summaries", [])
+
+        if len(change_sets) > 0:
+            change_set = change_sets[0]
```

Comment:
is the change set in index 0 always the latest, do we need to add any sorting to the service call?

Reply:
Just double checked, the change sets are sorted in ascending order if there are more than one. However, for the purposes of this PR we are going to delete the first change set and the template associated with it.

The others will be left to the user as we'll want to make sure the stack is deleted so that the customer is unblocked.

I'll make a change to add a log message to let them know that there may be lingering artifacts.